WalletWasabi/Crypto/ZeroKnowledge/ProofSystem.cs

```diff
@@ -211,11 +217,11 @@ public static Statement RangeProofStatement(GroupElement ma, IEnumerable<GroupEl
 			static int BitSquaredRow(int i) => BitRepresentationRow(i) + 1; // row for [ b*(B_i-Gg) - rb*Gh <=> b = b*b ] proof
 
 			// For each bit, add two equations and one term to the first equation.
-			var b = bitCommitments.ToArray();
-			for (int i = 0; i < bitCommitments.Count(); i++)
+			var negatedGh = Generators.Gh.Negate();
+			for (int i = 0; i < width; i++)
 			{
 				// Add [ -r_i * 2^i * Gh ] term to first equation.
-				equations[0, RndColumn(i)] = Scalar.Zero.CAddBit((uint)i, 1) * Generators.Gh.Negate();
+				equations[0, RndColumn(i)] = binaryVector[i] * negatedGh;
```

Comment:
these generators be precomputed statically up to the maximum width, since they do not depend on the runtime data in any way

Reply:
Done

Reply:
done